core/node/eth_sender/src/abstract_l1_interface.rs

```diff
@@ -206,34 +223,44 @@ impl AbstractL1Interface for RealL1Interface {
                     opt.max_fee_per_gas = Some(U256::from(base_fee_per_gas + priority_fee_per_gas));
                     opt.max_priority_fee_per_gas = Some(U256::from(priority_fee_per_gas));
                     opt.nonce = Some(tx.nonce.0.into());
-                    opt.transaction_type = if tx.blob_sidecar.is_some() {
-                        opt.max_fee_per_blob_gas = blob_gas_price;
-                        Some(EIP_4844_TX_TYPE.into())
-                    } else {
-                        Some(EIP_1559_TX_TYPE.into())
+                    opt.transaction_type = match operator_type {
+                        OperatorType::NonBlob => Some(EIP_1559_TX_TYPE.into()),
+                        OperatorType::Blob => Some(EIP_4844_TX_TYPE.into()),
```

Comment:
I've refactored this code, as it turns out in integration tests we actually don't send blobs, I will fix it (change so that we actually do send them) in a separate PR